### ISSUE TITLE
Cypress 3.0.2 fires up 400 and crashes the app

### ISSUE THREAD

**Issue by katerynae** [cypress-io/cypress#2210]
### Current behavior:

We have a form (we use Formik) that is checking the entered email and displays a modal if the email exist in our system. (React)

When the negative response is being returned from API cypress bubbles up 400 and the app crashes.
We updated the app on our side so the 400 doesn't break the app anymore but the issue persist in Cypress.

It was working for headless run with Electron for Cypress 3.0.1 and never worked with Chrome (We made sure we have "chromeWebSecurity": false and used plugin for Chrome 67) still didn't help

After updating to Cypress 3.0.2 it stopped working for Electron as well.

The weird issue that with DevTools open Cypress won't display the error and test keeps running.

### Steps to reproduce:
This is the error I see 

![image](https://user-images.githubusercontent.com/17116790/43224488-5d27b13a-9024-11e8-97b1-b72b7fdf25ab.png)

### Versions
Cypress 3.0.2

**Comment by jennifer-shehane** [comment]
Could you open Cypress using `DEBUG=cypress:* cypress open`, run the test that causes the crash and paste the logs that are printed here please?

**Comment by katerynae** [comment]
@jennifer-shehane I hope that will help because it's a lot.. I attached them in a doc file
[Logs.docx](https://github.com/cypress-io/cypress/files/2233200/Logs.docx)

**Comment by Bkucera** [comment]
@katerynae I've figured out the issue. We coerce `input[type=email]` to `type=text` since that is the only way to correctly read/write the selection on the input. However, we also cause a `blur` because that forces the element's state to be updated- this was causing an issue bc your app had an event handler on the `blur` event.

I've fixed the issue, and changes are in PR #2228 , slated for the next patch release, `3.0.3`
